### PR TITLE
Remove references to intialSync etc, and rewrite syncing

### PR DESCRIPTION
**Pull request by richvdh**
The 'Events' section of the C-S spec was full of outdated references to the v1
sync api, which is long dead. (https://matrix.org/jira/browse/SPEC-414).

Also, try to clarify why /sync distinguishes state and timeline
(https://matrix.org/jira/browse/SPEC-423).

### CODE REVIEW

File: specification/client_server_api.rst [hunk 1]
```diff
@@ -1046,7 +1047,9 @@ Leaving rooms
 A user can leave a room to stop receiving events for that room. A user must
 have been invited to or have joined the room before they are eligible to leave
 the room. Leaving a room to which the user has been invited rejects the invite.
-Once a user leaves a room, it will no longer appear on the |/initialSync|_ API.
+Once a user leaves a room, it will no longer appear in the response to the
+|/sync|_ API unless it is explicitly requested via a filter with the
+``include_leave`` field set to ``true``.
```
Comment (Kegsay) [inline specification/client_server_api.rst]: Should link to how you can remove it entirely (forget API)

File: specification/client_server_api.rst [hunk 2]
```diff
@@ -791,17 +777,32 @@ last request left off. Multiple events can be returned per long-poll.
...
 
+.. NOTE::
+
+  The ``/sync`` API returns a ``state`` list which is separate from the
+  ``timeline``. This ``state`` list allows clients to keep their model of the
+  room state in sync with that on the server. In the case of an intial
+  (``since``-less) sync, the ``state`` list represents the complete state of
+  the room at the **start** of the returned timeline (so in the case of a
+  recently-created room whose state fits entirely in the ``timeline``, the
+  ``state`` list will be empty).
```
Comment (Kegsay) [inline specification/client_server_api.rst]: This may warrant use of a rationale block to explain why `/sync` operates in this way.

**Comment by Kegsay** [comment]
LGTM aside from comments.

File: specification/client_server_api.rst [hunk 3]
```diff
@@ -791,17 +777,32 @@ last request left off. Multiple events can be returned per long-poll.
   being received (once per distinct API called). Clients SHOULD de-duplicate
   events based on the event ID when this happens.
 
+.. NOTE::
+
+  The ``/sync`` API returns a ``state`` list which is separate from the
+  ``timeline``. This ``state`` list allows clients to keep their model of the
+  room state in sync with that on the server. In the case of an intial
```
Comment (Kegsay) [inline specification/client_server_api.rst]: `intial`

File: specification/client_server_api.rst (hunk 2 above)
Comment (richvdh) [inline specification/client_server_api.rst]: You mean, why the state corresponds to the start of the timeline rather than the end?

I'm unconvinced that a rationale block would help. We're already in a 'note' block here, and to me it's kinda just the obvious behaviour.

File: specification/client_server_api.rst (hunk 3 above)
Comment (richvdh) [inline specification/client_server_api.rst]: done

File: specification/client_server_api.rst (hunk 1 above)
Comment (richvdh) [inline specification/client_server_api.rst]: The API in question is 3 lines below. I started trying to improve this, but it descended into a clarification of the whole section, so I'm going to make it a separate PR.

Comment (Kegsay) [inline specification/client_server_api.rst]: Okay.

File: specification/client_server_api.rst (hunk 2 above)
Comment (Kegsay) [inline specification/client_server_api.rst]: > why the state corresponds to the start of the timeline rather than the end?

Yes.

> it's kinda just the obvious behaviour.

Not good enough, *especially* since it wasn't always like this. We explicitly changed it to be the start rather than the end, and we should say why.

Comment (richvdh) [inline specification/client_server_api.rst]: fair enough

**Comment by richvdh** [comment]
PTAL

**Comment by Kegsay** [comment]
Wonderful explanation! LGTM
